pinot-common/src/main/java/org/apache/pinot/common/datablock/DataBlockUtils.java

```diff
@@ -230,6 +230,28 @@ public static ByteString toByteString(DataBlock dataBlock)
     return byteString;
   }
 
+  public static List<ByteString> toByteStrings(DataBlock dataBlock, int maxBlockSize)
+      throws IOException {
+    List<ByteBuffer> bytes = dataBlock.serialize();
+    if (bytes.isEmpty()) {
+      return List.of(ByteString.EMPTY);
+    }
+
+    List<ByteString> byteStrings = new ArrayList<>();
```

Comment:
Nit: it may be worth calculating the initial size of the array list. I think an upper bound would be to calculate the total byte size (iterating over `bytes`) and then add the number of elements in `bytes` (as in the worst case scenario, we add an extra small ByteString for each page in `bytes`)